osu.Game/Screens/Edit/Editor.cs

```diff
@@ -765,6 +775,17 @@ private void rebindClipboardBindables()
 
         protected void Copy() => currentScreen?.Copy();
 
+        protected void Duplicate()
+        {
+            // This is an initial implementation just to get an idea of how people used this function.
+            // There are a couple of differences from osu!stable's implementation which will require more work to match:
+            // - The "clipboard" is not populated during the duplication process.
+            // - The duplicated hitobjects are inserted after the original pattern (add one beat_length and then quantize using beat snap).
+            // - The duplicated hitobjects are selected (but this is also applied for all paste operations so should be changed there).
+            Copy();
+            Paste();
```

Comment:
The edge case of this easy solution is that if you:

1. Copy something to clipboard
2. Deselect all objects
3. Press the duplicate binding

it will do a paste. Not sure whether that should be fixed now. In theory there's a simple fix within reach (early-return if nothing is selected).

Reply:
Oops, I had this in an earlier version then refactored it out.